### DISCUSSION TITLE
Help on write mongo datasource object array query

### DISCUSSION THREAD

**Discussion by jackyshi3333** [getredash/redash#6952] [Q&A]
in MongoDB, one document schema define as below:
{
  "page_count": 7146,
  "page_number": 1,
  "page_size": 100,
  "total": 714598,
  "data": [
    {
      "id": "831333",
      "date": "1699364411806",
      "deadline_length_days": "22",
      "title": "Servicio de producción de células mesenquimales troncales adultas autólogas y alogénicas de medula ósea expandidas, vinculado al proyecto de investigación PIC18/00001"
        }
      ]
    }
question is how to query data.title and how to convert date field to time in redash, thanks.

**Comment by gaecoli** [comment]
url: https://xxxxx/
path: data
fields:[title]

**Reply by jackyshi3333** [reply]
sorry, I didn't fully got how to input this into query.
As the version of redash is 10.1.0, only json query pass check,
query content as this:
****
{
    "collection":"ProductPriceHistoryCopy",
            "url": "https://xxxxx/",
            "path": "data",
            "fields":["title"]
}

****
error message is as below:
****
Error running query: 'list' object has no attribute 'get'
****

Also below is tried without success:
****
{
    "collection":"ProductPriceHistoryCopy",
    "aggregation":[
        {
            "$project":
            {
            "url": "https://xxxxx/",
            "path": "data",
            "fields":["title"]
            }
        }
        ]
}
****
Anything wrong or missed, thanks again.

**Comment by gaecoli** [comment]
<img width="1179" alt="image" src="https://github.com/getredash/redash/assets/46274612/9f7f6c66-95cb-48fb-ab52-a8e3dca1210d">

**Reply by jackyshi3333** [reply]
Is it possible to show fields total and data[title], as mongodb unwind?

mongodb query unwind  as below:
****
db.ProductPriceHistory.aggregate([
        {"$unwind":"$data"},
        {"$project":
            {
                "total":"$total",
                "title":"$data.title"
            }
        }
    ])

****

**Reply by gaecoli** [reply]
Sorry, i mistake it. I thought it's json ds.

**Reply by jackyshi3333** [reply]
issue title updated，I thought mongo query is same as json ds
